packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -13,166 +13,189 @@ import {
+        .control${focusVisible} {
+            border-color: ${borderHoverColor};
+            box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
+        }
 
-    .control${focusVisible}::before {
-        outline: ${borderWidth} solid ${borderHoverColor};
-        outline-offset: -3px;
+        .control${focusVisible}::before {
+            outline: ${borderWidth} solid ${borderHoverColor};
+            outline-offset: -3px;
+        }
     }
 
-    .control:active::before {
-        outline: none;
-    }
+    @layer active {
+        .control:active {
+            box-shadow: none;
+            background-image: linear-gradient(
+                ${fillSelectedColor},
+                ${fillSelectedColor}
+            );
+            background-size: calc(100% - 2px) calc(100% - 2px);
+        }
 
-    :host([disabled]) .control::before {
-        box-shadow: none;
+        .control:active::before {
+            outline: none;
+        }
     }
 
-    .content {
-        display: contents;
-    }
+    @layer disabled {
+        :host([disabled]) {
+            color: ${buttonLabelDisabledFontColor};
+            cursor: default;
+        }
 
-    :host([content-hidden]) .content {
-        ${
-            /**
-             * Hide content visually while keeping it screen reader-accessible.
-             * Source: https://webaim.org/techniques/css/invisiblecontent/#techniques
-             * See discussion here: https://github.com/microsoft/fast/issues/5740#issuecomment-1068195035
-             */
-            ''
+        :host([disabled]) .control {
+            box-shadow: none;
+            background-image: none;
+            color: rgba(${actionRgbPartialColor}, 0.3);
         }
-        display: inline-block;
-        height: 1px;
-        width: 1px;
-        position: absolute;
-        margin: -1px;
-        clip: rect(1px, 1px, 1px, 1px);
-        clip-path: inset(50%);
-        overflow: hidden;
-        padding: 0;
-    }
 
-    [part='start'] {
-        display: contents;
-        ${iconColor.cssCustomProperty}: ${buttonLabelFontColor};
-    }
+        :host([disabled]) .control::before {
+            box-shadow: none;
+        }
 
-    :host([disabled]) slot[name='start']::slotted(*) {
-        opacity: 0.3;
-    }
+        :host([disabled]) slot[name='start']::slotted(*) {
+            opacity: 0.3;
+            ${iconColor.cssCustomProperty}: ${buttonLabelFontColor};
+        }
 
-    [part='end'] {
-        display: contents;
-        ${iconColor.cssCustomProperty}: ${buttonLabelFontColor};
+        :host([disabled]) slot[name='end']::slotted(*) {
+            opacity: 0.3;
+            ${iconColor.cssCustomProperty}: ${buttonLabelFontColor};
+        }
     }
 
-    :host([disabled]) slot[name='end']::slotted(*) {
-        opacity: 0.3;
+    @media (prefers-reduced-motion) {
+        .control {
+            transition-duration: 0s;
+        }
     }
 
     :host([content-hidden]) [part='end'] {
```

Comment:
I feel like instead of relying on styles outside of a layer having a higher importance we should instead make a layer named `top` and put it at the top of the layer order. An easier story to say every style needs to be in one of the layers.

Also if we have `top` maybe we should rename `default` to `base`.
`@layer base, hover, focusVisible, active, disabled, top;`

Base feels a little more appropriate than `default` because it has more styles than just default state styles, i.e. it has styles for the content-hidden state as well.

Also open to other naming ideas.